crates/model/src/order.rs

```diff
@@ -70,30 +53,39 @@ pub enum OrderStatus {
 
 impl Order {
     pub fn from_order_creation(
-        order_creation: &OrderCreation,
+        order: &OrderCreation,
         domain: &DomainSeparator,
         settlement_contract: H160,
         full_fee_amount: U256,
-        owner: H160,
         is_liquidity_order: bool,
-    ) -> Self {
-        Self {
+    ) -> Result<Self, VerificationError> {
+        let owner = order.verify_owner(domain)?;
+        Ok(Self {
             metadata: OrderMetadata {
-                creation_date: chrono::offset::Utc::now(),
                 owner,
-                uid: order_creation.uid(domain, &owner),
+                creation_date: chrono::offset::Utc::now(),
+                uid: order.data.uid(domain, &owner),
                 settlement_contract,
                 full_fee_amount,
                 is_liquidity_order,
                 ..Default::default()
             },
-            creation: *order_creation,
+            signature: order.signature,
+            data: order.data,
+        })
+    }
+
+    pub fn into_order_creation(self) -> OrderCreation {
+        OrderCreation {
+            data: self.data,
+            from: Some(self.metadata.owner),
+            signature: self.signature,
+            quote_id: None,
```

Comment:
nit: I guess the "usual" thing to do would be to implement `From<Order> for OrderCreation` and use the `.into()` function instead.
I don't feel strongly about this, though.

Reply:
Hard to say. I often prefer explicit functions because they're easier to understand.

Reply:
I added a `From` implementation (because I agree it makes sense) but kept the `into_order_creation` around because its useful in the many places (in e2e crate specifically) where type inference can't figure out what we want to convert into.